url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/Console.java

```diff
@@ -0,0 +1,623 @@
+        outputVariable1.addTerm(new Constant("f3", 0.040));
+        outputVariable1.addTerm(new Constant("f4", -0.180));
+        outputVariable1.addTerm(new Constant("f5", -0.190));
+        outputVariable1.addTerm(new Constant("f6", -0.040));
+        outputVariable1.addTerm(new Constant("f7", 0.090));
+        outputVariable1.addTerm(new Constant("f8", 0.120));
+        outputVariable1.addTerm(new Constant("f9", 0.040));
+        engine.addOutputVariable(outputVariable1);
+
+        OutputVariable outputVariable2 = new OutputVariable();
+        outputVariable2.setName("trueFx");
+        outputVariable2.setRange(-1, 1);
+        outputVariable2.setLockOutputRange(false);
+        outputVariable2.setLockValidOutput(true);
+        outputVariable2.setDefaultValue(Double.NaN);
+        outputVariable2.setDefuzzifier(new WeightedAverage());
+        outputVariable2.fuzzyOutput().setAccumulation(null);
+        outputVariable2.addTerm(Function.create("fx", "sin(inputX)/inputX", engine, true));
+        engine.addOutputVariable(outputVariable2);
+
+        OutputVariable outputVariable3 = new OutputVariable();
+        outputVariable3.setName("diffFx");
+        outputVariable3.setRange(-1, 1);
+        outputVariable3.setLockOutputRange(false);
+        outputVariable3.setLockValidOutput(false);
+        outputVariable3.setDefaultValue(Double.NaN);
+        outputVariable3.setDefuzzifier(new WeightedAverage());
+        outputVariable3.fuzzyOutput().setAccumulation(null);
+        outputVariable3.addTerm(Function.create("diff", "fabs(outputFx-trueFx)", engine, true));
+        engine.addOutputVariable(outputVariable3);
+
+        RuleBlock ruleBlock1 = new RuleBlock();
+        ruleBlock1.setName("");
+        ruleBlock1.setConjunction(null);
+        ruleBlock1.setDisjunction(null);
+        ruleBlock1.setActivation(new AlgebraicProduct());
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_1 then outputFx = f1", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_2 then outputFx = f2", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_3 then outputFx = f3", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_4 then outputFx = f4", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_5 then outputFx = f5", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_6 then outputFx = f6", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_7 then outputFx = f7", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_8 then outputFx = f8", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is NEAR_9 then outputFx = f9", engine));
+        ruleBlock1.addRule(Rule.parse("if inputX is any then trueFx = fx and diffFx = diff", engine));
+        engine.addRuleBlock(ruleBlock1);
+
+        return engine;
+    }
+
+    public static void exportAllExamples(String from, String to, String sourceBase, String targetBase) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.